### PR TITLE
fix(sidebar): close button styling

### PR DESCRIPTION
**Pull request by damianOfKyan**
### Proposed behaviour
Update the Toast close icon, when using a modern theme the icon will have a gold outline.
Screenshot 2020-03-04 at 12 06 24

### Current behaviour
The icon has a blue outline.
Screenshot 2020-03-04 at 11 53 59

### Checklist

- [x] Release notes (Conventional Commits) 
- [x] Unit tests
- [x] Cypress automation tests
<del>- [x] Storybook added or updated
<del>- [x] Theme support
<del>- [ ] Typescript `d.ts` file added or updated

### Additional context
There are other outstanding PR's related to close icon.
#2690 #2691 #2692 #2694 

### Testing instructions
1. `npm start`
1. Go to http://localhost:9001/?path=/story/sidebar--with-button

### CODE REVIEW

File: src/components/sidebar/sidebar.component.js [hunk 1]
```diff
@@ -17,33 +17,22 @@ class Sidebar extends Modal {
...
-            type='close'
-            tabIndex='0'
-            role='button'
-            onKeyDown={ this.onButtonKeyDown }
-          />
-        </SidebarCloseStyle>
-      );
-    }
-    return null;
+  closeIcon = () => {
```
Comment (dawidzarzycki) [inline src/components/sidebar/sidebar.component.js]: I suggest to replace this array method with normal class method just to be consistent in this class

File: src/components/sidebar/sidebar.component.js [hunk 2]
```diff
@@ -17,33 +17,22 @@ class Sidebar extends Modal {
...
+    if (!showCloseIcon || !onCancel) return null;
+    return (
+      <IconButton
+        onAction={ onCancel }
+      >
+        <Icon
+          type='close'
+          className='carbon-sidebar__close-icon'
+          data-element='close'
+          tabIndex='0'
```
Comment (dawidzarzycki) [inline src/components/sidebar/sidebar.component.js]: This ` tabIndex` should be removed. It causes double focus. One for the `IconButton` and the second one for the `Icon`

File: src/components/sidebar/sidebar.component.js [hunk 3]
```diff
@@ -17,33 +17,22 @@ class Sidebar extends Modal {
...
+  closeIcon = () => {
+    const { showCloseIcon, onCancel } = this.props;
+    if (!showCloseIcon || !onCancel) return null;
+    return (
+      <IconButton
+        onAction={ onCancel }
+      >
+        <Icon
+          type='close'
+          className='carbon-sidebar__close-icon'
```
Comment (dawidzarzycki) [inline src/components/sidebar/sidebar.component.js]: I think that this class is no more needed

Comment (damianOfKyan) [reply src/components/sidebar/sidebar.component.js]: fixed: fb7671f

File: src/components/sidebar/sidebar.component.js (hunk 2 above)
Comment (damianOfKyan) [reply src/components/sidebar/sidebar.component.js]: fixed: fb7671f

File: src/components/sidebar/sidebar.component.js (hunk 1 above)
Comment (damianOfKyan) [reply src/components/sidebar/sidebar.component.js]: fixed: fb7671f

File: cypress/features/regression/sidebar.feature [hunk 4]
```diff
@@ -5,6 +5,12 @@ Feature: Sidebar component
     Given I open "Sidebar" component page
       And I check open checkbox
 
+  @positive
+  Scenario: Check the cancel click event
+    When clear all actions in Actions Tab
+      And I close Sidebar
+    Then cancel action was called in Actions Tab
```
Comment (DlgSHi) [inline cypress/features/regression/sidebar.feature]:
```
@positive
   Scenario: Check the cancel click event
     Given clear all actions in Actions Tab
     When I close Sidebar
     Then cancel action was called in Actions Tab

File: cypress/features/regression/sidebar.feature [hunk 5]
```diff
@@ -5,6 +5,12 @@ Feature: Sidebar component
     Given I open "Sidebar" component page
```
Comment (DlgSHi) [inline cypress/features/regression/sidebar.feature]: also, we need to add a separate test for golden border

File: cypress/features/regression/sidebar.feature (hunk 4 above)
Comment (damianOfKyan) [reply cypress/features/regression/sidebar.feature]: fixed: 9af655a

File: cypress/features/regression/sidebar.feature (hunk 5 above)
Comment (damianOfKyan) [reply cypress/features/regression/sidebar.feature]: added: 9af655a

**Pull request merged**